src/openms/source/ANALYSIS/OPENSWATH/MRMAssay.cpp

```diff
@@ -1097,4 +1097,85 @@ void MRMAssay::detectingTransitionsCompound(OpenMS::TargetedExperiment& exp, int
     exp.setCompounds(compounds);
   }
 
+  void MRMAssay::filterUnreferencedDecoysCompound(OpenMS::TargetedExperiment &exp)
+  {
+    vector<TargetedExperiment::Compound> compounds = exp.getCompounds();
+    vector<std::string> descriptions_targets;
+    vector<std::string> descriptions_decoys;
+    vector<std::string> difference_target_decoys;
+    vector<std::pair<std::string, std::string>> reference_decoys;
+    vector<std::string> single_decoy_id;
+
+    for (const auto &it : compounds)
+    {
+      // extract potential target TransitionIds based on the decoy annotation
+      if (it.id.find("decoy") != std::string::npos)
+      {
+        String current_decoy = it.id;
+        String potential_target = std::regex_replace(current_decoy, std::regex("_decoy"), "");
```

Comment:
Yes, you are right, since the description is set as "_decoy" in the MetaboTargetedAssay based on the decoy annotation - one can assume that it is "_decoy" and the regex is not needed.